.github/workflows/tests.yml

```diff
@@ -11,24 +11,22 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        python-version: ["3.10", "3.11", "3.12"]
+        numpy-version: ["<2", ">=2"]
     runs-on: ubuntu-latest
 
     steps:
       - name: Checkout source
         uses: actions/checkout@v4
 
-      - name: Install poetry
-        run: pipx install poetry==1.8.2
-
       - name: Setup python
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-          cache: "poetry"
+          cache: "pip"
 
-      - name: Install dependencies
-        run: poetry install
+      - name: Install package
+        run: pip install "numpy${{ matrix.numpy-version }}" .[dev]
```

Comment:
Revert to using pip to install the package, because poetry does not give us the flexibility to run tests against different versions of numpy as far as I can make out.